controllers/argocd/project_controller.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2022 The KubeSphere Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package argocd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	"html/template"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"kubesphere.io/devops/pkg/api/devops/v1alpha3"
+	"kubesphere.io/devops/pkg/utils/k8sutil"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+// Reconciler is the reconciler of the DevOpsProject with Argo AppProject
+type Reconciler struct {
+	client.Client
+	log      logr.Logger
+	recorder record.EventRecorder
+}
+
```

Comment:
Is the configuration of RBAC ignored here?

Please refer to https://github.com/kubesphere/ks-devops/blob/c78b2f252e37030b6241ff53a2e48d258ee0a1e2/controllers/template/template_controller.go#L36-L37.

Reply:
Good catch!